janusgraph-core/src/main/java/org/janusgraph/core/JanusGraphFactory.java

```diff
@@ -344,7 +349,12 @@ private static ReadConfiguration getLocalConfiguration(File file) {
                 "Need to specify a readable configuration file, but was given: %s", file.toString());
 
         try {
-            PropertiesConfiguration configuration = new PropertiesConfiguration(file);
+            FileBasedConfigurationBuilder<PropertiesConfiguration> builder =
+                new FileBasedConfigurationBuilder<PropertiesConfiguration>(PropertiesConfiguration.class)
+                .configure(new Parameters().properties()
+                    .setFile(file)
+                    .setListDelimiterHandler(new DefaultListDelimiterHandler(',')));
```

Comment:
Creating a constant for the comma, or for the DefaultListDelimiterHandler?

Reply:
I was thinking for the DefaultListDelimiterHandler

Reply:
I prefer not. Reasons:
1) I think it does not bring much overhead because likely it won't be created many times or frequently. Likely one DefaultListDelimiterHandler instance per graph.
2) Although now this object is stateless and thread-safe, it could cause problems in the future in case it becomes stateful or non-thread-safe. The chance is very small, though.

Reply:
I don't think that's going to change in feature or patch releases. If that's changed, it's probably changed in a major (breaking change) release.
Right now the documentation clearly says:

> Implementation node: An instance of this class can safely be shared between multiple Configuration instances.

http://commons.apache.org/proper/commons-configuration/apidocs/org/apache/commons/configuration2/convert/DefaultListDelimiterHandler.html

I'm OK leaving it as is but just don't see a reason of creating unnecessary objects if one object is enough. 
That said, I'm OK leaving it as is if you think it will be better.

Reply:
Thanks, I didn't see that notice. Will change it.